File: ejercicios/interview practice/sortings/mergesort.c
#include <stdio.h>
#include <stdlib.h>

void mergesort(int *array, int len){
	if(len==1){
		array=array;
	}
	else if(len==2){
		
	}
	int half=len/2;
	int arre1[half];
	for(int i=0; i<half ; i++){
		arre1[i]=array[i];
	}
	int arre2[len-half];
	for(int i=half; i<(len-half) ; i++){
		arre1[i]=array[i];
	}
	mergesort(arre1);
	mergesort(arre2);
		
}

int main(){
	int array[9]={1,6,9,2,5,3,7,8,4};
	mergesort(array,9);	
	

}

File: ejercicios/interview practice/sortings/mergesort.py

def mergesort(alist):
	lar=len(alist)
	if(lar==1):
		return alist
	else:
		half=lar//2
		return merge(mergesort(alist[0:half])  ,  mergesort(alist[half:lar]))

def merge(a,b):
	ab=[]
	while(len(a)!=0 and len(b)!=0):
		if(len(a)==0  or len(b)==0):
			break
		elif(a[0]<b[0]):
			ab.append(a[0])
			del(a[0])
		else:
			ab.append(b[0])
			del(b[0])
	while(len(a)>0):
		ab.append(a[0])
		del(a[0])
	while(len(b)>0):
		ab.append(b[0])
		del(b[0])
	return ab


alist=[1,2,8,4,6,2,9,4,7,22]
print(mergesort(alist))
		
